### PR TITLE
[#16086] Messaging - Composer - Show current message for user context…

### LINKED ISSUE #16086: Messaging - Composer - Show current message for user context when responding
When a user selects a message to respond to, currently the user can not see which message they have selected to respond to.

There is some designs to help the user see which message they are responding to.

Implement this feature.

image2223212dsa

Designs:
https://www.figma.com/file/PPWkgOYlZZDxZv5SDGsZVV/Posts-%26-Attachments-for-Mobile?type=design&node-id=2869-258244&t=2WRn5madcEZpgWKx-0

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #16086

### CODE REVIEW

**Review by Parveshdhull — APPROVED** [review]

File: src/status_im2/contexts/chat/messages/content/view.cljs [hunk 1]
```diff
@@ -152,6 +150,14 @@
              (when @show-delivery-state?
                [status/status outgoing-status])])]]]))))
 
+(defn on-long-press
+  [message-data context keyboard-shown]
+  (rf/dispatch [:dismiss-keyboard])
```
Comment (OmarBasem) [inline src/status_im2/contexts/chat/messages/content/view.cljs:155]: Not for this PR, but what's the use of this event in the codebase? Isn't `rn/dismiss-keyboard!` sufficient?

**Comment by OmarBasem** [comment]
I think we need to animate the message component itself from its current position? Like when you long-press a message on WhatsApp or Telegram

File: src/status_im2/contexts/chat/messages/content/view.cljs (hunk 1 above)
Comment (flexsurfer) [reply src/status_im2/contexts/chat/messages/content/view.cljs:155]: effects shouldn't be used directly from view, event should be used instead

Comment (OmarBasem) [reply src/status_im2/contexts/chat/messages/content/view.cljs:155]: I believe this is not an effect: https://reactnative.dev/docs/keyboard#dismiss

**Comment by flexsurfer** [comment]
> I think we need to animate the message component itself from its current position? Like when you long-press a message on WhatsApp or Telegram

probably, we need requirements from design team

File: src/status_im2/contexts/chat/messages/content/view.cljs (hunk 1 above)
Comment (flexsurfer) [reply src/status_im2/contexts/chat/messages/content/view.cljs:155]: it is from re-frame pov, view must be pure

**Comment by qoqobolo** [comment]
@flexsurfer thanks for the fixes!

Question:
you've disabled any interaction with the content of the message as we agreed. But now, as a consequence, the content is not scrollable anymore. On small screens (e.g. on iPhone SE, Huawei) it is cut off. Do we accept this for now?
In this case, the scrollbar should be removed from this message (you can see it on Huawei in the video below).
Or is it possible to keep the ability to scroll, but block all other actions?

https://github.com/status-im/status-mobile/assets/67952253/0511a0d5-85e9-4a00-8188-6d318b0e59d2

---

Also, the first issue with link previews is still reproducible. Can replicate it on Android as well with certain links (e.g. https://www.youtube.com/watch?v=lSTYrMSkF7g&ab_channel=ThePsychedelicMuse)

Screenshot 2023-06-06 at 17 25 16

**Comment by flexsurfer** [comment]
> Or is it possible to keep the ability to scroll, but block all other actions?

it will be difficult to do, but we don't actually need full content, we just need to be sure for which message actions sheet is opened



> Also, the first issue with link previews is still reproducible.

should be fixed now, thanks

**Comment by qoqobolo** [comment]
Thanks for your work @flexsurfer!
PR is ready to be merged.

File: src/status_im2/contexts/chat/messages/content/view.cljs [hunk 2]
```diff
@@ -168,4 +175,5 @@
      [system-message-content message-data]
      [user-message-content message-data context keyboard-shown false])
    [reactions/message-reactions-row message-data
-    [user-message-content message-data context keyboard-shown true]]])
+    [rn/view {:pointer-events :none}
```
Comment (ibrkhalil) [inline src/status_im2/contexts/chat/messages/content/view.cljs:178]: Does clicking a mention and long pressing a message work with `:pointer-events none`?

File: src/status_im2/contexts/chat/messages/content/view.cljs [hunk 3]
```diff
@@ -152,6 +150,15 @@
              (when @show-delivery-state?
                [status/status outgoing-status])])]]]))))
 
+(defn on-long-press
+  [message-data context keyboard-shown]
+  (rf/dispatch [:dismiss-keyboard])
+  (rf/dispatch [:show-bottom-sheet
+                {:content       (drawers/reactions-and-actions message-data context)
+                 :selected-item (fn []
+                                  [rn/view {:pointer-events :none}
```
Comment (ibrkhalil) [inline src/status_im2/contexts/chat/messages/content/view.cljs:159]: Here too

File: src/status_im2/contexts/chat/messages/content/view.cljs (hunk 2 above)
Comment (flexsurfer) [reply src/status_im2/contexts/chat/messages/content/view.cljs:178]: this is for selected-item , it should be inactive

File: src/status_im2/contexts/chat/messages/content/view.cljs (hunk 3 above)
Comment (flexsurfer) [reply src/status_im2/contexts/chat/messages/content/view.cljs:159]: same

File: src/status_im2/contexts/chat/messages/content/view.cljs (hunk 2 above)
Comment (ibrkhalil) [reply src/status_im2/contexts/chat/messages/content/view.cljs:178]: Totally, Great work!

**Pull request merged**